### ISSUE TITLE
Simple Motion Effects

### ISSUE THREAD

**Issue by iamandrewpeters** [beaverbuilder/feature-requests#28]
**Is your feature request related to a problem? Please describe.**
I know this is a bells and whistles thing. But the ability to have motion effects would be huge. Scrolling effects and mouse over effects would open many design options.  

**Describe the solution you'd like**
I would like to see an 'enable motion effects' option in the main BB settings. I would love modules to have the ability to be positioned accordingly and have various popular options for scrolling, motion and mouseover effects (transparency, blur, rotate, tilt, scale, etc). 

I would also love to load scripts conditionally. I understand that animation js can add some weight to the page. So only having those loaded on pages that make use of them would be helpful. 

**Describe alternatives you've considered**
The only alternatives I've considered and use are Ginger Soul's Plugins. I tried Elementor on a site that I was asked to do motion stuff for, and it was annoying to use. I switched back to BB quickly. haha. 

**Additional context**

**Comment by DavidWaumsley** [comment]
I wonder if this could be solved with tutorials?  This seems to have come in and out of fashion Obviously it also has some usability issues and perhaps some SEO ones too. 

I think those earlier 2013/14 scripts when this thing treading before are not used now (Elementor created their own and I think much of the load issues is with the added CSS that has to be in place for it)

I just wonder in the age where things like Lottie are appearing that BB could end up with something baked in to core that will be little used and prone to go out of date.  I wonder how many still use BB's parallax backgrounds now.

 If long term use is likely to be low I feel it should be a plugin.  That's said if my fears over this were alleviated it would be nice to have.

**Comment by ccollabgist** [comment]
> I wonder if this could be solved with tutorials? This seems to have come in and out of fashion Obviously it also has some usability issues and perhaps some SEO ones too.
> 
> I think those earlier 2013/14 scripts when this thing treading before are not used now (Elementor created their own and I think much of the load issues is with the added CSS that has to be in place for it)
> 
> I just wonder in the age where things like Lottie are appearing that BB could end up with something baked in to core that will be little used and prone to go out of date. I wonder how many still use BB's parallax backgrounds now.
> 
> If long term use is likely to be low I feel it should be a plugin. That's said if my fears over this were alleviated it would be nice to have.

Honestly, I'm at the point where I'd like to see them have some of those bells and whistles vs. none. BB is a great tool but they still need to keep competitive. I hardly ever do anything with vanilla BB modules and mainly use UABB or PP because of those extra features - I rarely if ever touch the vanilla modules. :(

Oh how I've yearned for tasteful hover animations (and submitted feature requests for them).

While some things might be trends, I think others are likely to stick around and I'd rather them start adding some of the funner features than none so that I can stop hearing my colleagues and clients talking about Elementor this and Elementor that.

**Comment by gbissland** [comment]
I'm voting for this also, I think a bump is needed as well in BB, BUT I wouldn't want it 'all on' by default.

It is a real fine line. I think I'd like to see established js animations added rather than anything homebaked. I haven't really researched any, so got no suggestions.

I think it has taken three years of using BB to find the right mix and balance of pagebuilder and our own js/css. How much of each to use, for performance and time to make jobs more profitable.
I am actually finding we're using the 3rd party addons less the longer I use BB. Just using some PP modules now and love the innovation coming from their team. I don't use SoulSections as much as I should.

Same with the theme customiser. Not using that now. Starting with all defaults as CSS/child theme has made projects a lot cleaner and faster.

Whether the target 'developer' level BB users would use the extra animations I'm not sure, but I think they would appeal to newer users and cashflow is good.

@DavidWaumsley I used a parallax background this week ... what are you saying?  😉

**Comment by DavidWaumsley** [comment]
@gbissland  Me too!

**Comment by Pross** [comment]
> @gbissland Me too!

Thumbs up the original post, that's what gets counted if you want to _me too_ anything.
https://share.getcloudapp.com/eDu14qwb

**Comment by DavidWaumsley** [comment]
@Pross  Yeah I know. I was replying to Gareth regarding his comment to me on parallax background . Sorry, it's probably best not to have much conversation here, but as the original post was a good suggestion I wanted to explain reasons for thinking now may not be the time to commit something like this to core.

**Comment by brandontan903** [comment]
Metoo

**Comment by pixelloop** [comment]
I would definitely like to see this in BB.

I think BB has been left in the dust visually speaking a bit over the years by alternatives.

**Comment by ccollabgist** [comment]
> I would definitely like to see this in BB.
> 
> I think BB has been left in the dust visually speaking a bit over the years by alternatives.

Completely agree. It's definitely the most stable page builder IMO but barely ever mentioned these days vs alternatives. It's lagging so far behind others it makes me a sad panda. :(

**Comment by pixelloop** [comment]
> Completely agree. It's definitely the most stable page builder IMO but barely ever mentioned these days vs alternatives. It's lagging so far behind others it makes me a sad panda. :(

Yea, people only talk about Elementor and Oxygen, etc, and BB is often forgotten. 

Webflow also, they incorporated the same sort of visual features and it's become one of its biggest selling points.

Would love BB to catch up a bit in that department. I think these should be core features that can be turned off in the settings for people who don't need any of them.

**Comment by AcceleratedWebSolutions** [comment]
I've recently used a theme called VamTam Builder and it's used with BB.  The animations had much more control than BB.  See my screenshots.  

I'm hoping BB will come out with more control and features associated with their animations.  I've been tempted to try Elegant Themes Divi to get more types of animation and more control.

<img width="953" alt="Screen Shot 2021-03-08 at 9 40 29 AM" src="
<img width="632" alt="Screen Shot 2021-03-08 at 9 41 01 AM" src="https://user-images.githubusercontent.com/15944331/110378270-63b5c680-8023-11eb-954a-8d4d74ea864b.png">
https://user-images.githubusercontent.com/15944331/110378265-631d3000-8023-11eb-9141-8ed01da4879a.png">
<img width="650" alt="Screen Shot 2021-03-08 at 9 40 38 AM" src="https://user-images.githubusercontent.com/15944331/110378267-63b5c680-8023-11eb-8b45-b3ddbe5f4601.png">
<img width="451" alt="Screen Shot 2021-03-08 at 9 40 46 AM" src="https://user-images.githubusercontent.com/15944331/110378268-63b5c680-8023-11eb-8045-04144e64aea9.png">

**Comment by NikoNice44** [comment]
My agency has been using beaver builder and the beaver builder theme since 2015, for over 300 websites. But this is what could make us switch. We really need this for our more “modern” clients.

**Comment by jomoweb** [comment]
Vote up for this feature! My clients are increasingly asking for more motion. They really want scroll animations like rellax.js or animateonscroll.js. We've been adding js libraries manually, but they don't play nice with the builder. A parralax animation will animate while in the builder and cannot click the edit icons on elements.

We've looked into Elementor's motion library, and that is a fairly ideal solution for what we need. But I really don't want to have to switch from BB. I love how the page builder works and have been working with it for 5 years.

A plugin would be fine. I have looked at the existing add ons like power pack and uabb, but they do not offer this.

**Comment by gbissland** [comment]
Agreed and voted.

@jomoweb  You should check out what Dave @gingersoulrecords is building at https://soulplugins.co/

The Soul Magic 2 plugin adds extra animations including I think some of the Greensock animation library into BB with premade components and examples to drag in and get started.

It is aimed more at devs so you need to play code in the editor but his aim is to teach as well.

**Comment by jomoweb** [comment]
@gbissland - Thanks for the tip. SoulMagic looks like it could fill the void! I wish I would have known about it a few projects ago :)

**Comment by gingersoulrecords** [comment]
Hey, thanks @gbissland!

@jomoweb I'm happy to chat with you about SoulMagic2. Some hot stuff going on.
